### PR TITLE
Fix libkrw tfp0 and switch libkrw-dopamine repos

### PR DESCRIPTION
**Pull request by Cryptiiiic**
### All Submissions

* [x] Have you made sure there aren't any other open [Pull Requests](https://github.com/ProcursusTeam/Procursus/pulls) for the same update/change?
* [ ] This Pull Request doesn't contain any package additions; it's a small change (e.g README change)

### Package Additions/Updates

* [x] Have you confirmed this builds & works as intended on an iOS device (if applicable)?
* [ ] Have you confirmed this builds & works as intended on a macOS device (if applicable)?

### CODE REVIEW

**Review by asdfugil — CHANGES_REQUESTED** [review]
1. This was missed last time, but please remove the extra line above `# libkrw-dopmaine.mk Sign`
2. The control file should depend on `firmware (<= 16.6.1)`, not 15.5
3. Name the folder in build_misc as `libkrw-dopamine`, since libjailbreak is too generic and does not tell much about which makefile uses it

File: build_patch/libkrw/tfp0.patch [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
-+++ b/src/libkrw.c
-@@ -107,9 +107,7 @@ static void iterate_plugins(int (*callback)(void *), void **check) {
+--- a/src/libkrw.c	2024-04-20 14:46:12.796173476 -0700
++++ b/src/libkrw.c	2024-04-20 18:01:14.500961452 -0700
+@@ -160,3 +160,8 @@
+     if (krw_handlers.physwrite == NULL) return ENOTSUP;
+     return krw_handlers.physwrite(from, to, len, granule);
  }
++
++__attribute__((visibility("hidden")))
```
Comment (asdfugil) [inline build_patch/libkrw/tfp0.patch:8]: Why declare this function at all? Remove it and its usage entirely.
You may want to fork libkrw though.

Comment (Cryptiiiic) [reply build_patch/libkrw/tfp0.patch:8]: The original code has it as hidden. Meh I didn't really change anything codewise so I doesn't need a fork imo.

**Comment by Cryptiiiic** [comment]
> 1. This was missed last time, but please remove the extra line above `# libkrw-dopmaine.mk Sign`
> 
>     2. The control file should depend on `firmware (<= 16.6.1)`, not 15.5
> 
>     3. Name the folder in build_misc as `libkrw-dopamine`, since libjailbreak is too generic and does not tell much about which makefile uses it
1. will be done
2. Oops that's fixed locally already, not pushed
3. How bout `dopamine2-libjailbreak` in case future packages also link it? Or should I just stick to `libkrw-dopamine`?

**Comment by asdfugil** [comment]
dopamine2-libjailbreak is fine by me just don’t be libjailbreak

**Comment by asdfugil** [comment]
Actually I prefer libjailbreak-dopamine2

**Comment by Cryptiiiic** [comment]
> Actually I prefer libjailbreak-dopamine2

Done

**Review by asdfugil — APPROVED** [review]

**Pull request merged**